sklearn/cluster/optics_.py

```diff
@@ -194,7 +194,7 @@ class OPTICS(BaseEstimator, ClusterMixin):
        the Conference "Lernen, Wissen, Daten, Analysen" (LWDA) (2018): 318-329.
     """
 
-    def __init__(self, min_samples=5, max_eps=np.inf, metric='minkowski', p=2,
+    def __init__(self, min_samples=5, max_eps=np.inf, metric='euclidean', p=2,
```

Comment:
The default metric `minkowski` doesn't accept the sparse matrix.

See https://github.com/huntzhan/scikit-learn/blob/5f7e6de69e425cfdfeeb82cf692b81fe22589dd5/sklearn/metrics/pairwise.py#L1546-L1555

Reply:
But this won't work if the user changes `p`. So I'd rather reinterpret the metric as euclidean below when using the metric, rather than changing the default

Reply:
make sense.

Reply:
Are you able to add commits to resolve it?

Reply:
Yeah, I will resolve this issue and add test cases this week.